platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/cc/impl/problems/CommonReport.kt

```diff
@@ -126,7 +127,13 @@ class ConfigurationCacheReport(
             val hashingStream = HashingOutputStream(Hashing.md5(), spoolFile.outputStream().buffered())
 
             private
-            val writer = HtmlReportWriter(hashingStream.writer(), htmlReportTemplate)
+            val hashingWriter: Writer = hashingStream.writer()
+
+            private
+            val jsonModelWriter: JsonModelWriter = JsonModelWriter(JsonModelWriterCommon(hashingWriter))
```

Comment:
🤔  We don't seem to need these fields in the class itself, and only need those values to create the final writer.
Should we create a function that creates the `HtmlReportWriter`?